### PR TITLE
Update Drupal Core, Simplify and Update Dev Dependencies...

### PR DESCRIPTION
**Pull request by jacine**
This PR has a number of fixes for Composer, in preparation for Composer 2.  See details in diff comments (coming shortly). Composer itself has been updated to the latest version, so as to prevent it from auto-updating to version 2, as we have dependencies that are not ready for it. Doing so resolves this warning: 

> Warning from <https://repo.packagist.org>: You are using an outdated version of Composer. Composer 2.0 is about to be released and the older 1.x releases will self-update directly to it once it is released. To avoid surprises update now to the latest 1.x version which will prompt you before self-updating to 2.x. 

Until these packages are updated, we will not be able to use Composer 2:

- `html2text/html2text` (required by `drupal/core`)
- `mandrill/mandrill` (required by `drupal/mandrill`)
- `easyrdf/easyrdf` (required by `drupal/sendgrid_integration`)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -108,29 +106,24 @@
...
         "drupal/core-dev": "^8.8",
         "drupal/devel": "^1.2",
         "drupal/drupal-extension": "^3.1",
         "drupal/search_kint": "^1.0",
-        "drupal/stage_file_proxy": "^1.0@alpha",
-        "drupal/twig_xdebug": "^1.0",
-        "jcalderonzumba/gastonjs": "^1.0.2",
-        "jcalderonzumba/mink-phantomjs-driver": "^0.3.1",
-        "mikey179/vfsstream": "^1.2",
-        "phpunit/phpunit": "^6.5"
```
Comment (jacine) [inline composer.json:133]: We do not need to require any of these anymore! 🎉 

The [`drupal/core-dev`](https://github.com/drupal/core-dev) package takes care of everything we [need](https://github.com/drupal/core-dev/blob/8.9.x/composer.json).

File: composer.json [hunk 2]
```diff
@@ -108,29 +106,24 @@
...
+        "drupal/stage_file_proxy": "^1.1",
+        "drupal/twig_xdebug": "^1.0"
```
Comment (jacine) [inline composer.json:126]: Just some updates to packages we already have.

File: composer.json [hunk 3]
```diff
@@ -153,16 +146,12 @@
         "drupal-scaffold": "DrupalComposer\\DrupalScaffold\\Plugin::scaffold",
         "prepare-for-pantheon": "DrupalProject\\composer\\ScriptHandler::prepareForPantheon",
         "post-install-cmd": [
-            "@drupal-scaffold",
             "DrupalProject\\composer\\ScriptHandler::createRequiredFiles"
         ],
         "post-update-cmd": [
-            "DrupalProject\\composer\\ScriptHandler::createRequiredFiles",
-            "find .circleci/scripts/pantheon/ -type f | xargs chmod 755",
-            "find tests/scripts/ -type f | xargs chmod 755"
```
Comment (jacine) [inline composer.json:162]: All the scaffold related commands are no longer needed here.  The `pantheon-systems/drupal-integrations` and `drupal/core-composer-scaffold` packages take care of all of this now.

File: composer.json [hunk 4]
```diff
@@ -250,10 +230,7 @@
     },
     "config": {
         "optimize-autoloader": true,
-        "preferred-install": {
-            "sf-digital-services/sfgov-pattern-lab": "source",
```
Comment (jacine) [inline composer.json:254]: With this library gone, we no longer need this 🎉

File: web/.gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+/.eslintrc.json
+/INSTALL.txt
+/README.txt
+/example.gitignore
```
Comment (jacine) [inline web/.gitignore:4]: All of these new `.gitignore` files come [drupal/core-composer-scaffold](https://github.com/drupal/core-composer-scaffold).

**Review by jacine — COMMENTED** [review]
Hi @aekong!  Left you a ton of comments to make reviewing this easier.

**Review by aekong — APPROVED** [review]
thank you for updating this!

**Pull request merged**